### PR TITLE
DUI3-71 receive bindings and receive geometry

### PR DESCRIPTION
**Pull request by connorivy**
#### Connectors / Revit
- Added receive binding (pretty much copied from Rhino)
- Modified service registrations for new types
  - Added default traversal function. @JR-Morgan, is this correct for dui3?
- Registered a run on thread method that uses the Revit Async package
  - Running on main thread alone is not sufficient
- Added simple host object "builder" that just opens a transaction, traverses, receives, and then commits the transaction
- Copied transaction manager from dui2 connector

#### Converter / Revit
- added import method next to export method on referencePointConverter (and interface)
- Changed name of 'RevitRootToHostConverter' which implements 'IRootToSpeckleConverter' to accurately reflect the implemented interface (aka replaced ToHost with ToSpeckle)
- Added toHost scaling service to mirror toSpeckle scaling service

#### Converter / Revit / ToHost
- implemented ITypedConverter interface for several types of geometries
- added a modelCurve topLevel converter to be able to actually use the typedConverters

#### Converter / Revit / ToSpeckle
- I moved a bunch of files into this 'toSpeckle' folder to mirror the folder structure of Rhino

#### Converter / Rhino
- Weird random change in a rhino file. I'm thinking this is just something weird going on with git. Need to ask @adamhathcock if this is a legit change

## To-do before merge:

- [ ] ask @JR-Morgan about revit traversal method
- [ ] ask @adamhathcock about rhino file change

## Validation of changes:

receiving works for model lines of various types

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/DependencyInjection/RevitConnectorModule.cs [hunk 1]
```diff
@@ -49,20 +50,24 @@ public void Load(SpeckleContainerBuilder builder)
...
     builder.AddScoped<IRootObjectBuilder<ElementId>, RevitRootObjectBuilder>();
-
-    // register send conversion cache
     builder.AddSingleton<ISendConversionCache, SendConversionCache>();
+
+    // receive operation and dependencies
+    builder.AddScoped<ReceiveOperation>();
+    builder.AddScoped<IHostObjectBuilder, RevitHostObjectBuilder>();
+    builder.AddScoped<TransactionManager>();
+    builder.AddSingleton(DefaultTraversal.CreateTraversalFunc());
```
Comment (JR-Morgan) [inline DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/DependencyInjection/RevitConnectorModule.cs:71]: Yep perfect. This new and improved `DefaultTraversal` is exactly what we should be using (injecting as you have here) from now on.
Tldr, this one should behave functionally equivalent to the old one, while no longer being dependent on the old `CanConvertFunction`.

**Comment by adamhathcock** [comment]
The PR is definitely wrong somehow as it has a lot of commits from me in it but should be covered with a merge.  Let me know if you need help redoing/rebasing the branch?

**Comment by connorivy** [comment]
closing this pr in favor of a new one that doesn't have a bunch of Adam's commits in it https://github.com/specklesystems/speckle-sharp/pull/3542
